monitor/penalty.go

```diff
@@ -20,17 +19,23 @@ func (m *Monitor) processPenaltyQueue() {
 	if !m.isPusher() {
 		return
 	}
-
+	var keys, vals [][]byte
+	m.dbLock.RLock()
 	iterator, err := m.db.Iterator(PenaltyKeyPrefix, storetypes.PrefixEndBytes(PenaltyKeyPrefix))
 	if err != nil {
 		log.Errorln("Create db iterator err", err)
 		return
 	}
-	defer iterator.Close()
-
 	for ; iterator.Valid(); iterator.Next() {
-		event := NewPenaltyEventFromBytes(iterator.Value())
-		err = m.db.Delete(iterator.Key())
+		keys = append(keys, iterator.Key())
+		vals = append(vals, iterator.Value())
+	}
+	iterator.Close()
```

Comment:
why not defer close?

Reply:
because the iterator is not used anymore

Reply:
actually you are right, there could be err return before this. will change back to defer close

Reply:
oops sorry, I misread the code, there is no return before the close, expect the err when creating the iterator. So the current code is good and will be kept